### ISSUE TITLE
[Bug]: Qwen3-Embedding-0.6B三种请求方式，数值结果都不一致

### ISSUE THREAD

**Issue by DankoZhang** [vllm-project/vllm#27256]
### Your current environment

vllm=0.9.2

### 🐛 Describe the bug

vllm serve方式启动：
`CUDA_VISIBLE_DEVICES=1 vllm serve */Qwen/Qwen3-Embedding-0___6B \
  --task embedding \
  --host 0.0.0.0 \
  --port 8002 \
  --dtype auto \
  --gpu-memory-utilization 0.2 \
  --max-num-seqs 8 \
  --max-model-len 8192 \
  --trust-remote-code`

vllm本地加载：
```python
LLM(model=model_path, task="embed", gpu_memory_utilization=0.8, max_model_len=8192, enforce_eager=is_debug)
```

qwen3-embedding原生transformers加载模型方式：
```python
AutoModel.from_pretrained(model_path, device_map="auto").eval()
```

上面三种方式加载向量化模型，输入如下语句，向量化结果不同，这是什么原因？

![Image](https://github.com/user-attachments/assets/bf768779-e0dd-4439-93ef-81f7e9c1ed82)

![Image](https://github.com/user-attachments/assets/e90286e7-1358-4f94-a0a9-f309eb80522c)

<img width="502" height="60" alt="Image" src="https://github.com/user-attachments/assets/a1356660-fa50-4538-a029-8e120cefd08f" />

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

**Comment by noooop** [comment]
sim = F.cosine_similarity(
            torch.tensor(embeddings_0), torch.tensor(embeddings_1), dim=0
        )

你试试 cosine_similarity， fp16 在 0,01 以内，fp32 要精度更高一点，但非常慢
